sunpy/coordinates/transformations.py

```diff
@@ -61,6 +63,48 @@
            'hme_to_gei', 'gei_to_hme', 'gei_to_gei']
 
 
+# Global variable to control debugging output
+_debug_output = False
+# Global counter to keep track of layers of transformations
+_layer_level = 0
+
+
+def _transformation_debug(description):
+    """
+    Decorator to produce debugging output for a transformation function: its description, inputs,
+    and output.  Unicode box-drawing characters are used.
+    """
+    def decorator(func):
+        @wraps(func)
+        def wrapped_func(*args, **kwargs):
+            global _layer_level
```

Comment:
does this need to be global global or once per transform function, if the latter you could put in in the `def decorator()` scope?

Reply:
The level counter needs to be shared across all transformation functions to be able to keep track of what layer of transformation you are in, so I'm pretty sure it wouldn't work at the `def decorator()` scope.  If C-style static variables were a thing in Python, that'd be the best way to do this.  Instead, that kind of persistence requires either a global variable or a class variable, and the latter just seemed unnecessarily clunky from a readability point of view.

Reply:
ok that makes some sense. I think this is fine as is then.